source/installation/from-source/core-apps.rst

```diff
@@ -0,0 +1,36 @@
+.. _core_apps:
+
+Install OnDemand Core Applications
+==================================
+
+  .. note::
+
+    The OnDemand team is in the process of moving core applications to a monolithic repository which will significantly simplify installation.
+
+OnDemand's core applications are stored in their own Github repositories and should be cloned to ``/var/www/ood/apps/sys/$APP``:
+
+  - `ood-activejobs`_: ``/var/www/ood/apps/sys/activejobs``
+  - `ood-dashboard`_: ``/var/www/ood/apps/sys/dashboard``
+  - `ood-fileeditor`_: ``/var/www/ood/apps/sys/file-editor``
+  - `ood-fileexplorer`_: ``/var/www/ood/apps/sys/files``
+  - `ood-myjobs`_ : ``/var/www/ood/apps/sys/myjobs``
+  - `ood-shell`_: ``/var/www/ood/apps/sys/shell``
+
+  .. _ood-activejobs: https://github.com/OSC/ood-activejobs/
+  .. _ood-dashboard: https://github.com/OSC/ood-dashboard/
+  .. _ood-fileeditor: https://github.com/OSC/ood-fileeditor/
+  .. _ood-fileexplorer: https://github.com/OSC/ood-fileexplorer/
+  .. _ood-myjobs: https://github.com/OSC/ood-myjobs/
+  .. _ood-shell: https://github.com/OSC/ood-shell/
+
+Each application has its own dependencies which need to be installed (from either NPM or Ruby Gems) by running the following:
+
+  .. code-block:: sh
+
+    cd /var/www/ood/apps/sys/$APP
+    # We have both Node and Rails applications, let's just cover both in a single command
+    sudo NODE_ENV=production RAILS_ENV=production scl enable rh-nodejs6 rh-ruby24 rh-git29 -- bin/setup
```

Comment:
If we document one step as using SCL should we document the `yum` command necessary to install those software collections? Maybe in the system-dependencies page?

Reply:
Think it would be `yum install rh-nodejs6-npm rh-ruby24-ruby-devel rh-git29-git`. Usually pulling in `-devel` also pulls in runtime.

Reply:
Yes. I would think that installing from source on a CentOS6 or CentOS7 box would mean using yum install and rpms for the dependencies that are non OnDemand code, and then git cloning the ondemand repos.

Which means, perhaps, the ondemand scl package should be used in the install from source? And if thats the case would the command be `scl enable ondemand -- bin/setup` instead of `scl enable rh-nodejs6 rh-ruby24 rh-git29 -- bin/setup`

Reply:
> ... A use case for some folks wanting this was 'hacking around', I'm not sure how this addresses that, but I'm also not sure _how_ to address that.

My preferred way to do that sort of thing is to spin up a VM using RPMs for OnDemand, then replace the apps that I want to hack on with symlinks to something that is convenient for me to edit locally. The caveat with my workflow with VirtualBox is that the apps do not work if their `vendor/bundle | node_modules` directories are not on the guest. So there is additional symlink-ing necessary and things get messy.

That workflow addresses how people can get a version of OnDemand to hack around with, but isn't satisfying to people who cannot use RPMs.

Reply:
> If we document one step as using SCL should we document the `yum` command necessary to install those software collections? Maybe in the system-dependencies page?

I was trying to avoid becoming too vendor specific. I can flip this the other way and call out specific `yum` package names and versions when necessary.

As for changing the SCL command to use the `ondemand` SCL I can do that, but I'll still call out the specific items that must be in the `$PATH` for non-RPM users.